suprb/optimizer/rule/mutation.py

```diff
@@ -160,3 +161,52 @@ def min_percentage(self, rule: Rule, random_state: RandomState):
         bounds = rule.match.bounds
         bounds[:, 0] -= random_state.uniform(0, self.sigma[0], size=bounds.shape[0])
         bounds[:, 1] += random_state.uniform(0, self.sigma[1], size=bounds.shape[0])
+
+
+class AdaptiveMutation(RuleMutation):
+    """Start off by using a given operator until the fitness decreased
+       more than self.number_of_worse_iterations times. Then change the mutation operator
```

Comment:
This should only work with & or , replacement strategies, shouldn't it? So we should probably note that here. Or maybe even better, don't require the fitness to have decreased but rather not to have increased? That way this would work for + ES as well (albeit likely with a different tolerance required)